### PR TITLE
Region for 3s tiles fails if not multiple of 3s

### PR DESCRIPTION
**Pull request by PaulWessel**
See [form post](https://forum.generic-mapping-tools.org/t/error-pygmt-gmtcliberror-module-grdimage-failed-with-status-code-78/829) for details. This PR simply adds a failing test.

### CODE REVIEW

**Comment by PaulWessel** [comment]
Note: I thought I had fixed that since it was also reported [earlier](https://forum.generic-mapping-tools.org/t/gradient-problem-using-earth-relief-03s/787) but I cannot find the pull request of issue.  Can you, @seisman?

**Comment by PaulWessel** [comment]
Note it works without -I but fails with -I.

**Comment by seisman** [comment]
This one? https://github.com/GenericMappingTools/gmt/issues/3206

**Comment by PaulWessel** [comment]
Thanks yes.  OK, so this current case does not involve 15s so perhaps takes a different path.  I will debug.

**Comment by PaulWessel** [comment]
Fixed by #4130.

**Comment by seisman** [comment]
But do we want to add the test for it?

**Comment by PaulWessel** [comment]
Ooops, yes, this PR has the test script and now the correct PS.

**Comment by seisman** [comment]
One concern about this PR is, this PR uses earth_relief_03. It means we have to cache the earth_relief_03s grid (at least some tiles).

**Comment by PaulWessel** [comment]
Might we have a flag to prevent a test from being run in the CI, only locally?

**Comment by PaulWessel** [comment]
Likewise, perhaps remove backport?

**Comment by PaulWessel** [comment]
Before adding another single flag, perhaps thing about deeper about test partitioning.  Ways to consider subsets of tests:

```
Ability to skip or select tests that are CPU/time intensive
Ability to skip or select tests that require large data sets
Ability to skip or select tests only for the API (the api dir)
Ability to select tests only for one module (i.e., only the test/module folder)
```

Might be others.  Hopefully the flag (maybe this is a bitflag) can help the CI know what to pre-download? A key issue for us is often the time it takes.  I made a histogram of run times for the tests once.  It is long-tailed, with mostly quick tests and a few slow ones.  A linear scheme does not seem sensible, maybe a flag 1, 2 , 3 that maps to log10 of some scaled time?

**Comment by seisman** [comment]
> Ability to skip or select tests only for the API (the api dir)
> Ability to select tests only for one module (i.e., only the test/module folder)

It's already supported. `ctest -R api` to run api tests only, and `ctest -R psxy` to run psxy-specific tests.

Currently, tests have inconsistent names (e.g., `GMT_hammer.sh` in `doc/scripts` directory, `ex01/ex01.sh` in `doc/examples` directory, and `pshistogram/flip.sh` in `test` directory). In #3949, I plan to let all tests use their relative path as the test name. For example, the test name should be `doc/scripts/GMT_hammer.sh` instead of `GMT_hammer.sh`. After #3949 is done, it's possible to split all tests into three groups: examples, scripts in the documentation, and the real tests.

File: test/grdimage/rounding.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+# Failing script while fixing https://forum.generic-mapping-tools.org/t/error-pygmt-gmtcliberror-module-grdimage-failed-with-status-code-78/829
+# Now reported as issue ###
```
Comment (seisman) [inline test/grdimage/rounding.sh:3]: Please fix/improve the comments.

**Comment by seisman** [comment]
For this test, I think we can run `gmt grdcut -R3:57/4:18/44:00/44:15 @earth_relief_03s -G/dev/null` to cache the two tiles (~3 MB) in the CI.

**Comment by PaulWessel** [comment]
Like this?

File: test/grdimage/rounding.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+# Failing script while fixing https://forum.generic-mapping-tools.org/t/error-pygmt-gmtcliberror-module-grdimage-failed-with-status-code-78/829
+# Now reported as issue https://github.com/GenericMappingTools/gmt/pull/4130
+
+# Ensure the two tiles are downloaded first:
+gmt grdcut -R3:57/4:18/44:00/44:15 @earth_relief_03s -G/dev/null
```
Comment (seisman) [inline test/grdimage/rounding.sh:6]: No. I didn't mean this. It should be done in the CI script:

https://github.com/GenericMappingTools/gmt/blob/66f5bdce05d4e34ecc371633da95d358f517dd57/ci/azure-pipelines-mac.yml#L80-L90

Comment (PaulWessel) [reply test/grdimage/rounding.sh:6]: OK, I agree. Should I or you move that code to the .yml file then?

Comment (seisman) [reply test/grdimage/rounding.sh:6]: I'll do that.

**Pull request merged**